src/rgw/rgw_asio_frontend.cc

```diff
@@ -69,6 +69,9 @@ class StreamIO : public rgw::asio::ClientIO {
                                           yield[ec]);
     if (ec) {
       ldout(cct, 4) << "write_data failed: " << ec.message() << dendl;
+      if (ec==boost::asio::error::broken_pipe) {
+        stream.lowest_layer().shutdown(tcp::socket::shutdown_both, ec);
```

Comment:
we're throwing `ec` as a `rgw::io::Exception` just below, but this call to `shutdown()` will mutate `ec`. maybe use a separate variable `boost::system::error_code ec_ignored` for the call to `shutdown()`?

Reply:
the intention was to override the `ec` to indicate the specific condition has occured
for example the error printed to the log at
https://github.com/ceph/ceph/blob/4425f3e678155c3fa0cb6f0b6f4c9e3866415c14/src/rgw/rgw_process.cc#L292
changes from
`7fb47eedc700  0 ERROR: client_io->complete_request() returned Broken pipe`
to 
`7fc0e14da700  0 ERROR: client_io->complete_request() returned Transport endpoint is not connected`

it is not essential, changing to `ec_ignored`